lib/src/configuration.rs

```diff
@@ -0,0 +1,59 @@
+#[derive(Debug, Clone)]
+pub struct Configuration {
```

Comment:
Please provide doc comments for this and each other struct in this PR (and in the future), containing some or all of:
- The purpose of the struct
- What each field represents
- Any other notes (special considerations or constraints)